features/net/network-socket/nsapi_types.h

```diff
@@ -148,6 +148,18 @@ typedef enum nsapi_option {
 } nsapi_option_t;
 
 
+/*  C linkage definition
+ *
+ *  Attribute garuntees emitted function has C linkage.
+ *  Available in both C and C++
+ */
+#ifdef __cplusplus
+#define NSAPI_C_LINKAGE extern "C"
```

Comment:
Got it. In that case, is a general EXTERN in toolchain not the way to handle this? Would guess this is more useful than just NSAPI